File: kakao/문자열 압축.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 1000, cnt = 0, tmp = 0;
    string str;
    
    for(int i = 1; i <= s.size() / 2 + 1; i++)
    {
        cnt = 1;
        tmp = 0;
        str = s.substr(0, i);

        for(int j = i; j < s.size(); j += str.size())
        {
            if(str == s.substr(j, i))
                cnt++;
                
            else
            {
                if(cnt == 1)
                    tmp += str.size();
                else if(cnt < 10)
                    tmp += (str.size() + 1);
                else if(cnt < 100)
                    tmp += (str.size() + 2);
                else
                    tmp += (str.size() + 3);
                str = s.substr(j, i);
                cnt = 1;
            }
        }
        if(cnt == 1)
            tmp += str.size();
        else if(cnt < 10)
            tmp += (str.size() + 1);
        else if(cnt < 100)
            tmp += (str.size() + 2);
        else
            tmp += (str.size() + 3);
        if(tmp < answer)
            answer = tmp;
    }
    
    
    return answer;
}

File: kakao/문자열 압축.js
function solution(s) {
    let answer = s.length;
    
    for(let i = 1; i <= s.length / 2 + 1; i++){
        let count = 1;
        let tmp = 0;
        let str = s.substr(0, i);
        
        for(let j = i; j < s.length; j += str.length){
            if(str === s.substr(j, i)){
                count++;
            }
            else{
                if(count === 1){
                    tmp += str.length;
                }
                else{
                    tmp += count.toString().length + str.length;
                }
                str = s.substr(j, i);
                count = 1;
            }
        }
        
        if(count === 1){
            tmp += str.length;
        }
        else{
            tmp += count.toString().length + str.length;
        }
        
        answer = answer < tmp ? answer : tmp;
    }
    
    return answer;
}